packages/markitdown/src/markitdown/converters/_pdf_converter.py

```diff
@@ -65,13 +64,30 @@ def convert(
                     extension=".pdf",
                     feature="pdf",
                 )
-            ) from _dependency_exc_info[
-                1
-            ].with_traceback(  # type: ignore[union-attr]
+            ) from _dependency_exc_info[1].with_traceback(  # type: ignore[union-attr]
                 _dependency_exc_info[2]
             )
 
-        assert isinstance(file_stream, io.IOBase)  # for mypy
-        return DocumentConverterResult(
-            markdown=pdfminer.high_level.extract_text(file_stream),
-        )
+        cur_pos = file_stream.tell()
+        markdown = pdfminer.high_level.extract_text(file_stream)
+        if markdown.strip() == "":
+            # Try to leverage LLM OCR capabilities when PDF is not searchable
+            llm_client = kwargs.get("llm_client")
+            llm_model = kwargs.get("llm_model")
+            if llm_client and llm_model:
+                file_stream.seek(cur_pos)
+                llm_prompt = """You are an advanced document extraction AI. Your task is to analyze the provided
+                document, understand its content and context, and produce a perfectly structured Markdown document
+                from the text within it. Do not translate neither generate new text. Retain the structure of the
+                original content, ensuring that sections, titles, headers and important details are clearly separated.
+                If the image contains any tables, lists and code snippets format them correctly to preserve their
+                original meaning. Only a valid Markdown-formatted output is allowed."""
+                markdown = llm_caption(
+                    file_stream,
+                    stream_info,
+                    client=llm_client,
+                    model=llm_model,
+                    prompt=llm_prompt,
+                )
+
+        return DocumentConverterResult(markdown=str(markdown))
```

Comment:
There is an issue of PDFs containing both mineable text and images that contain text. It would be nice to have a more sophisticated branching mechanism that accounts for this and/or allowing an API to override by the markitdown caller.

Reply:
Are you thinking on something like replacing the usage of `extract_text` with `extract_pages` and iterate over its non-text elements, like `LTImage` and `LTFigure`?

Layout system reference:
https://pdfminersix.readthedocs.io/en/latest/topic/converting_pdf_to_text.html#topic-pdf-to-text-layout